src/pages/RepoPage/CoverageTab/OverviewTab/subroute/CoverageChart/CoverageChart.tsx

```diff
@@ -0,0 +1,138 @@
+import { format } from 'date-fns'
+import { useParams } from 'react-router-dom'
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from 'recharts'
+
+import { useBranches } from 'services/branches'
+import { useRepoOverview } from 'services/repo'
+import {
+  ChartConfig,
+  ChartContainer,
+  ChartLegend,
+  ChartLegendContent,
+  ChartTooltip,
+  ChartTooltipContent,
+} from 'ui/Chart'
+
+import { useBranchSelector, useRepoCoverageTimeseries } from '../../hooks'
+
+const formatDate = (date: string) => format(new Date(date), 'MMM d, yyyy')
+
+const Placeholder = () => (
+  <div
+    data-testid="coverage-chart-placeholder"
+    className="h-[250px] min-h-[200px] w-full animate-pulse rounded bg-ds-gray-tertiary xl:h-[380px]"
+  />
+)
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+function CoverageChart() {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const { data: overview } = useRepoOverview({
+    provider,
+    repo,
+    owner,
+  })
+  const { data: branchesData } = useBranches({ repo, provider, owner })
+  const { selection } = useBranchSelector({
+    branches: branchesData?.branches ?? [],
+    defaultBranch: overview?.defaultBranch ?? '',
+  })
+
+  const { data, isPreviousData, isLoading, isError } =
+    useRepoCoverageTimeseries({
+      branch: selection?.name,
+      options: {
+        enabled: !!selection?.name,
+        suspense: false,
+        keepPreviousData: true,
+      },
+    })
+
+  if (!isPreviousData && isLoading) {
+    return <Placeholder />
+  }
+
+  if (isError) {
+    return (
+      <div className="flex min-h-[250px] items-center justify-center xl:min-h-[380px]">
+        <p className="text-center">The coverage chart failed to load.</p>
+      </div>
+    )
+  }
+
+  if (data?.measurements.length < 2) {
```

Comment:
Same comment here. IMO showing a flat line for one data point may be good.

Reply:
See other PR for response =P